Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs

```
@@ -1786,6 +1786,98 @@ public async Task TestScriptCreateOnContainerRecreateFromDifferentPartitionKeyPa
+                    Document readDocument = (await container.ReadItemAsync<Document>(document.Id, pKey)).Resource;
+                    Assert.AreEqual(document.ToString(), readDocument.ToString());
+                }
+
+                //Document Update.
+                foreach (ItemResponse<Document> obj in documents)
+                {
+                    Cosmos.PartitionKey pKey = new Cosmos.PartitionKey(new object[] { obj.Resource.GetValue<string>("ZipCode"), obj.Resource.GetPropertyValue<string>("Address") });
+                    Document document = (await container.ReadItemAsync<Document>(obj.Resource.Id, pKey)).Resource;
+                    document.SetPropertyValue("Name", document.Id);
+                    Document readDocument = (await container.ReplaceItemAsync<Document>(document, document.Id, pKey)).Resource;
+                    Assert.AreEqual(readDocument.GetValue<string>("Name"), document.GetValue<string>("Name"));
+                }
+
+                //Document Delete.
+                foreach (Document document in documents)
+                {
+                    Cosmos.PartitionKey pKey = new Cosmos.PartitionKey(new object[] { document.GetValue<string>("ZipCode"), document.GetPropertyValue<string>("Address") });
+                    Document readDocument = (await container.DeleteItemAsync<Document>(document.Id, pKey)).Resource;
+                    try
+                    {
+                        readDocument = await container.ReadItemAsync<Document>(document.Id, pKey);
+                    }
+                    catch (CosmosException clientException)
+                    {
+                        Assert.AreEqual(clientException.StatusCode, HttpStatusCode.NotFound);
+                    }
+                }
+
+            }
+            catch (Exception)
+            {
+                Assert.Fail();
```

No need for this catch. Just let the exception bubble up and fail the test.